src/main/java/io/beanmapper/dynclass/DynamicClassGeneratorHelper.java

```diff
@@ -0,0 +1,56 @@
+package io.beanmapper.dynclass;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import javassist.CtClass;
+
+public class DynamicClassGeneratorHelper {
+
+    private DynamicClassGeneratorHelper() {}
+
+    /**
+     *
+     * @param ctClass
+     * @return
+     */
+    public static Class<?> getNeighbourClass(CtClass ctClass) {
+
+        Class<?> neighbourClass = null;
+
+        try (InputStream inputStream = ClassLoader.getSystemResourceAsStream(
+                ctClass.getPackageName().replaceAll("[.]", "/"))) {
+            try (InputStreamReader isReader = new InputStreamReader(inputStream)) {
+                try (BufferedReader bufferedReader = new BufferedReader(isReader)) {
+                    neighbourClass = bufferedReader.lines()
+                            .filter(line -> line.endsWith(".class"))
+                            .findFirst()
+                            .map(line -> loadClass(ctClass.getPackageName(), line))
+                            .orElseThrow(ClassNotFoundException::new);
+                }
+            }
+        } catch (IOException e) {
+            // TODO Handle Exception
+            throw new RuntimeException(e);
+        } catch (ClassNotFoundException e) {
+            // TODO Handle Exception
+            throw new RuntimeException(e);
+        }
+
+        return neighbourClass;
+    }
+
+    private static Class<?> loadClass(String packageName, String className) {
+        Class<?> loadedClass = null;
+        try {
```

Comment:
Weer dependant op hoe je errors will teruggeven, maar deze zou je ook aan de method als throws declaration kunnen toevoegen en hem meenemen in je try catch er boven.

![image](https://user-images.githubusercontent.com/105443742/188852372-3b6e4ab9-bd48-461c-8363-8971529df3fc.png)